### ISSUE TITLE
pypi.org shows latest version, pip see's previous version

### ISSUE THREAD

**Issue by martinohanlon** [pypa/warehouse#3962]
**Describe the bug**
I have recently uploaded a new version of the guizero package 0.5.1 (https://pypi.org/project/guizero/) which is being displayed as available on pypi.org.

![image](https://user-images.githubusercontent.com/2329501/39981518-c530a5c6-5748-11e8-971e-2eb4cbb759a0.png)

However pip is not reporting the new version as being available, only the previous 0.5.0.

![image](https://user-images.githubusercontent.com/2329501/39981556-edd352ee-5748-11e8-85de-b5cf1677a58b.png)

I have tried across multiple platorms (windows, mac, linux) all with the same result.

I have tried upgrading pip (already latest version), with no effect.

As far as I can tell everything has been packaged correctly and pypi.org shows no errors.

**Expected behavior**
I am expecting guizero 0.5.1 to be available to install via pip.

**To Reproduce**

Run:

```
pip3 install guizero
```

or 

```
pip search guizero
```

v0.5.0 is reported as the latest version.

**My Platform**
I have tested on:
+Windows 10, pip version 10.0.1
+MacOS, pip version 10.0.1
+Raspbian, pip version 9.0.1

**Comment by martinohanlon** [comment]
Some additional information, the new vesion was uploaded using twine v1.11.0, setuptools 38.5.1.

![image](https://user-images.githubusercontent.com/2329501/39983034-2d7101da-574e-11e8-86da-6839263a550a.png)

**Comment by martinohanlon** [comment]
And now pip is reporting that version 0.5.1 is the latest...

```
C:\Users\Martin O'Hanlon-LT>pip3 search guizero
guizero (0.5.1)  - Python module to allow learners to easily create GUIs
  INSTALLED: 0.5.1 (latest)
```

Timing issues? Propagation of data? Something else?

**Comment by ewdurbin** [comment]
Hi @martinohanlon, thanks for the report.

Our search index is not updated incrementally yet, instead it is refreshed once every 3 hours. This would certainly lead to the behavior you're reported.

As such, I believe this is a duplicate of the feature request we are tracking in #701.

**Comment by martinohanlon** [comment]
@ewdurbin Its certainly seems to be a symptom of not having #701 .

I suppose if this is currently considered to be normal behaviour...

**Comment by ewdurbin** [comment]
We certainly want to implement incremental search index updates! Initial attempts in #3693 seemed to aggravate another issue, #3746

The pycon sprints are happening over the next four days and search issues are something we are going to try to prioritize.

**Comment by di** [comment]
To be clear, since `pip` uses the `/simple` API, which is instantaneously updated, you should be able to `pip install` your package as soon as you have uploaded it.

It's only `pip search` which depends on the search index, which is currently updated every 3 hours (as @ewdurbin said).

**Comment by martinohanlon** [comment]
I just did a couple of tests on this because what @di said didnt match with my experience when I created this issue.

This is what I did / found:
- I created a new packaged and uploaded it to pypi
- As soon as I uploaded it, as @di said, I was able to install it
- I then incremented the version number and uploaded it to pypi
- At this point I was unable to install the new version using `pip install package --upgrade` as it only found the previous version
- I uninstalled the packaged using `pip uninstall package` and tried to install it, but again it would only find the previous version
- I was only able to install the new package by disabling the cache using `pip --no-cache-dir install package` option

In summary, providing you have never installed the packaged before, or you disable the cache, you can install a package simultaneously, otherwise its dependent on the search index being updated.

**Comment by di** [comment]
I think what you're hitting here is that pip's local HTTP cache is not stale enough to go out and actually hit PyPI the second time. I don't think any point you're interacting with the search index.

This might be worth raising on the pip issue tracker, but at first glance I don't see any way to avoid this without using `--no-cache-dir` as it's just a side effect of caching.

**Comment by rschwiebert** [comment]
@ewdurbin This is happening for me right now for `mimeparse==0.1.4`, which appears on pypi having been released Jan 2, 2013, and yet I'm having trouble with it in 2019. `pip==19.1` reports 0.1.2 and 0.1.3 as being available, but it doesn't list or allow me to install 0.1.4.

Also, `pip search` *does* appear to be aware of 0.1.4.

Could some old packages be affected by a different problem?

**Comment by jamadden** [comment]
`mimeparse 0.1.4` has no files on PyPI (note the absence of the "Download files" link on the right of https://pypi.org/project/mimeparse/0.1.4/), so pip has no way to install it. 

I can see how the search result would be a bit confusing.

**Comment by caramdache** [comment]
I have the issue with `data-importer 3.1.1`. It can be see by `pip3 search`, but not by `pip3 install` which only sees `data-importer 3.0.2`.

**Comment by jamadden** [comment]
`data-importer 3.1.1` *only* provides [a Python 2 wheel](https://pypi.org/project/data-importer/3.1.1/#files). In contrast, 3.0.2 *only* provides a source distribution. Python 3 will not be able to install a Python 2 wheel, so 3.1.1 will not be used. 3.0.2's source distribution can be used by any version of Python so that's what Python 3 will be stuck with.

**Comment by caramdache** [comment]
Thanks @jamadden, I've reported this to the misconfigured package.

**Comment by valdergallo** [comment]
@caramdache I made the upload for p3k today. Thanks for helping me @jamadden and @caramdache

**Comment by amandal1810** [comment]
Python 3.8.3 (64-bit), Windows 10

pip cannot see latest version from pypi.org
![image](https://user-images.githubusercontent.com/8520454/83054112-7ef4ae00-a06f-11ea-8c24-fbe6edc6d93c.png)

screenshot from pypi.org
![image](https://user-images.githubusercontent.com/8520454/83054154-93d14180-a06f-11ea-9196-02abc00006e4.png)

**Comment by di** [comment]
@amandal1810 This is likely due to https://github.com/protocolbuffers/protobuf/issues/7520
